lib/data/repositories/repository_provider.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:farmsmart_flutter/data/repositories/FlameLink.dart';
+import 'package:farmsmart_flutter/data/repositories/article/ArticleRepositoryInterface.dart';
+import 'package:farmsmart_flutter/data/repositories/article/implementation/ArticlesRepositoryFlamelink.dart';
+import 'package:farmsmart_flutter/data/repositories/article/implementation/MockArticlesRepository.dart';
+import 'package:farmsmart_flutter/data/repositories/plot/PlotRepositoryInterface.dart';
+import 'package:farmsmart_flutter/data/repositories/plot/implementation/MockPlotRepository.dart';
+import 'package:farmsmart_flutter/flavors/app_config.dart';
+import 'package:flutter/material.dart';
+
+
+class RepositoryProvider {
+  final BuildContext context;
+
+  FlameLink cms;
+  bool isMockData;
```

Comment:
Just thinking instead of the bool, could we make the Repository provider an interface (abstract) and then have a mock and a firebase implementation? this approach is fine as we have the repo interfaces to isolate, but wanted to discuss is that a step too far.

Reply:
DONE